File: examples/scheduling/cron-manage.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Inspect and manually fire cron triggers from the client.
export async function manageCronTriggers(client: JsBaoClient, triggerId: string) {
  // #region example
  const { items } = await client.cronTriggers.list();
  const trigger = await client.cronTriggers.get(triggerId);
  await client.cronTriggers.test(triggerId); // fire once, now
  // #endregion example
  return { items, trigger };
}

File: examples/scheduling/cron-manage.swift
import JsBaoClient

// Inspect and manually fire cron triggers from the client.
func manageCronTriggers(client: JsBaoClient, triggerId: String) async throws {
  // #region example
  let list = try await client.cronTriggers.list()
  let items = list.items
  let trigger = try await client.cronTriggers.get(triggerId: triggerId)
  _ = try await client.cronTriggers.test(triggerId: triggerId) // fire once, now
  // #endregion example
  _ = (items, trigger)
}
